modules/ROOT/examples/live-demos/custom-toolbar-menu-button-search/index.js

```diff
@@ -0,0 +1,53 @@
+tinymce.init({
+  selector: 'textarea#custom-toolbar-menu-button-search',
+  height: 500,
+  toolbar: 'mybutton',
+
+  setup: (editor) => {
+    /* Menu items are recreated when the menu is closed and opened, so we need
+       a variable to store the toggle menu item state. */
+    let toggleState = false;
+
+    /* example, adding a toolbar menu button */
+    editor.ui.registry.addMenuButton('mybutton', {
+      text: 'My searchable button',
+      search: {
+        placeholder: 'What do you want to search for?'
+      },
+      fetch: (callback, fetchContext) => {
+        if (fetchContext.pattern.length > 0) {
+          callback([
+            {
```

Comment:
Perhaps an actual search would have been a better example. This example's intent though is to demonstrate that you can use pattern however you like.